composer/core/surgery.py

```diff
@@ -142,3 +187,113 @@ def count_module_instances(model: torch.nn.Module, module_class: Type[torch.nn.M
         count += count_module_instances(child, module_class)
 
     return count
+
+
+def _tensor_in(tensor: torch.Tensor, iterable: Iterable[torch.Tensor]):
+    """Returns whether `tensor is element` for any element in `iterable`
+    This function is necessary because `tensor in iterable` does not work
+    reliably for `Tensor`s.
+    See https://discuss.pytorch.org/t/how-to-judge-a-tensor-is-in-a-list/15998/4
+    for further discussion.
+    """
+    return any(tensor is elem for elem in iterable)
+
+
+def _find_param_in_optimizer(param: torch.nn.parameter.Parameter, optimizer: torch.optim.Optimizer) -> int:
+    """Returns the index of the optimizer ``param_group`` containing ``param``
+    Optimizers store their parameters within an iterable of ``dict``s called
+    :attr:`~torch.optim.Optimizer.param_groups`.
+    By default, there is only one group in :attr:`~torch.optim.Optimizer.param_groups`
+    that containing all the parameters, but there can be more than one. This
+    function is a simple utility to identify which parameter group in
+    :attr:`~torch.optim.Optimizer.param_groups` contains a given parameter, if any. The information
+    might be desirable to, e.g., inspect the optimizer settings being used
+    for a given parameter, or to remove unused parameter tensors from
+    the optimizer.
+
+    Args:
+        param (torch.nn.parameter.Parameter): The parameter to search for.
+        optimizer (torch.optim.Optimizer): The optimizer to search within.
+    
+    Returns:
+        int: The index within `opt.param_groups` of the first group containing ``param``.
+
+    Raises:
+        RuntimeError: If the ``param`` is not in the ``opt`.
```

Comment:
Probably better to return -1 instead of throwing here; more consistent with e.g., string `find()`. Doesn't matter much since it's private, but it made reading the `test_optimizer_surgery_removed_params_gone` a little confusing. Although maybe there was some reason I (or you?) did it this way that I'm missing

Reply:
Yup, I did refactor it -- I prefer to raise exceptions than to use return codes, but just switched it back -- agree it makes sense to be similar to `find`